### PR TITLE
docs: :memo: add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by limwa**
This pull request adds contributing instructions to the repository.
The main focus points are that the repository adopts the Git Flow workflow and the use of Conventional Commits.

Let me know what you think.

### CODE REVIEW

**Review by gwealm — DISMISSED** [review]

**Review by Naapperas — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+First of all, thank you for your interest in contributing to this repository! :rocket:
+
+However, before you create your pull request (or commit), there are a few things you need to know!
+
+## Contributing
+We all know how frightening it is contributing for the first time to a repository. Don't worry, though!
```
Comment (limwa) [inline CONTRIBUTING.md]:
```suggestion
We all know how frightening contributing for the first time to a repository is. Don't worry, though!
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   
+   This is also know as the [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) workflow
+   and helps ensure that the repository remains organized.
+   
+6. Make your changes
+7. Stage the files you've changed for commit using `git add <files>`
+8. Commit your changes using `git commit`
+
+   In this repository, we follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+   *Your contribution **will not** be accepted if your commits do not follow said specification.*
```
Comment (limwa) [inline CONTRIBUTING.md]: I need to check if this is correctly formated before I approve this pull request.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+   That said, you can use [gitmoji](https://gitmoji.dev/) in your commit message to spice it up a bit.
+   An acceptable commit message with gitmoji in it would look something like `docs(readme): :memo: add instalation instructions`.
+   
+9. Push your changes to your fork using `git push`
+10. Go to your fork's Github page, select the branch you've created and start a pull request
+
+    The base for your pull should be the `develop` branch.
+
+If you've completed these steps, you should now have an open pull request in this repository.
+Now, you can sit back, relax, and wait for someone in the [@MIEIC-FEUP/quiz-trainer-core](https://github.com/orgs/MIEIC-FEUP/teams/quiz-trainer-core) team to review your pull request.
```
Comment (limwa) [inline CONTRIBUTING.md]:
```suggestion
Now, you can sit back, relax, and wait until three members of the @MIEIC-FEUP/quiz-trainer-core team review your pull request.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+If you are asked to make some changes and you don't know how to do them, don't be afraid to ask! We are here to help you!
+
+## TL;DR for experienced users
+In this repository, we enforce the use of the [Git Flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) workflow and the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+If your changes do not cumply with any of those, your contribution will be rejected.
```
Comment (limwa) [inline CONTRIBUTING.md]:
```suggestion
If your changes do not comply with any of those, your contribution will be rejected.
```

**Review by limwa — COMMENTED** [review]
Overall, I still need to fix some typos.

**Comment by limwa** [comment]
I know I'm being a bit hypocrite by not using Conventional Commits on these last commits, but I am on mobile and it doesn't let me write the commit messages. So, instead of merging normally, we should instead squash and merge!

**Review by Naapperas — DISMISSED** [review]

**Review by gwealm — DISMISSED** [review]

**Review by gwealm — APPROVED** [review]

**Review by Naapperas — APPROVED** [review]

**Pull request merged**